### PR TITLE
UI feedback for owner/curator workspace delete conditions

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadinstitute.atlassian.net/browse/GAWB-2352
Requires complementary Orchestration PR: https://github.com/broadinstitute/firecloud-orchestration/pull/500

## Changes:
Update the delete workspace logic to include un-publishing from Library if necessary.

## Testing:
* Delete a normal, un-published workspace and ensure no regressions
* Delete a published workspace and ensure that it does not show up in Library.
* Try to delete a published workspace with users that are not allowed.
   * Workspace "READER"
   * Workspace "OWNER" but not "CATALOG" permissions
   * "CATALOG" user, but not "OWNER"

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 1]
```diff
@@ -240,12 +242,17 @@
...
+                   publisher? (and curator? (or catalog-with-read? owner?))]
+               [comps/SidebarButton {:style :light :margin :top :color (if isLocked :text-lighter :exception-state)
+                                     :text "Delete" :icon :delete
+                                     :data-test-id "delete-workspace-button"
+                                     :disabled? (cond
+                                                  isLocked "This workspace is locked."
+                                                  (not (and published? publisher?)) "Curator permissions insufficient to delete this workspace.")
+                                     :onClick #(modal/push-modal
+                                                [DeleteDialog {:workspace-id workspace-id
+                                                               :published? published?}])}]))]}]]))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs]: You can collapse this a bit with a restructure:
```clojure
[DeleteDialog (utils/restructure workspace-id published?)]
```

**Review by MattPutnam — APPROVED** [review]
👍

**Review by davidangb — APPROVED** [review]
:+1: looks good to me, assuming it passes PO review for wording.

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 2]
```diff
@@ -29,15 +29,17 @@
...
      [comps/OKCancelForm
       {:header "Confirm Delete"
        :content
        [:div {}
         (when (:deleting? @state)
           [comps/Blocker {:banner "Deleting..."}])
         [:p {:style {:margin 0}} "Are you sure you want to delete this workspace?"]
-        [:p {} "Bucket data will be deleted too."]
+        [:p {} (str "Deleting it will delete the associated bucket data"
+                    (when (:published? props) " and unpublish the workspace from the Data Library")
```
Comment (rushtong) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs:41]: Bucket deletion language change was requested by comms and PO.

File: src/cljs/main/broadfcui/common/components.cljs [hunk 3]
```diff
@@ -178,7 +179,11 @@
                       :borderRadius 5}
               :data-test-id (:data-test-id props)
               :data-test-state (if disabled? "disabled" "enabled")
-              :onClick (if disabled? (create-error-message disabled?) (:onClick props))}
+              :onClick (if disabled?
+                         (if disabled-handler
+                           (disabled-handler)
+                           (create-error-message disabled?))
+                         (:onClick props))}
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/components.cljs]: I don't like this approach. Also, if we're mucking with modals we should be converting to the new modal.

**Review by davidangb — APPROVED** [review]
:+1: works for me; I defer to Matt on the modal handling

File: src/cljs/main/broadfcui/common/components.cljs (hunk 3 above)
Comment (rushtong) [reply src/cljs/main/broadfcui/common/components.cljs]: I will look for the new modal stuff and re-implement.

**Comment by MattPutnam** [comment]
👍

**Review by davidangb — APPROVED** [review]
:+1:

**Pull request merged**